### ISSUE TITLE
[components-autocomplete--default] Auto Complete dropdown result providing incorrect results when "No Results" case

### ISSUE THREAD

**Issue by rafiqdeen** [tyler-technologies-oss/forge#514]
**Describe the bug:**
When searching a term that provide No Results in the dropdown then clicking at the end of the term is providing default dropdown results which looks like the dropdown showing incorrect result for the searched term.

**To Reproduce:**
1. Go to https://forge.tylerdev.io/main/?path=/docs/components-autocomplete--default
2. Click on "Choose State" Input field and Enter "xyz"
3. Again click inside the Input field.
4. Then seeing default search results in the dropdown.

**Expected behavior:**
If the term doesn't have any results, show a warning "No Results found" or Preserve the empty dropdown even after clicking the input field second time if the search term have no results.

**Screenshots:**
![Screenshot from 2024-04-01 12-45-32](https://github.com/tyler-technologies-oss/forge/assets/9208269/5bc4e82c-40e5-4430-a3d6-b9c3e1cc32db)

**Please complete the following information:**
 - Forge version: [e.g. v2.0.0]
 - I have searched existing issues before creating this report? [Y/N]
 - Browser: [e.g. Chrome, Firefox, Safari, All]
 - Platform: [e.g. desktop, mobile, All]
 - OS: [e.g. Windows, Mac, Linux, iOS, Android, All]

**Comment by DRiFTy17** [comment]
@rafiqdeen Yeah I think this is a bug... Thanks for reporting it! The autocomplete shouldn't attempt to execute the filter when clicking on the field if it is already focused. We'll get this taken care of so that it doesn't call the `filter` callback in this scenario.

Additionally, at this point the autocomplete is not responsible for what appears in the dropdown as far as showing "No results found". We will leave this up to developers for now, but I do think it would be worth us considering adding this as an configuration option to make it easier in the future. Would you be able to add this idea to our Canny board [here](https://tylerforge.canny.io/feature-requests)? This will allow others to vote on the feature request and provide discussion.

**Comment by rafiqdeen** [comment]
Thank you. Added this feature request in Canny board : https://tylerforge.canny.io/feature-requests/p/prevent-auto-complete-dropdown-providing-incorrect-results
